### ISSUE TITLE
Potential typo when computing `primal_feasibility_in_rhs_0` in `global_primal_residual`?

### ISSUE THREAD

**Issue by oswinso** [Simple-Robotics/proxsuite#254]
I think maybe there is a typo when computing `primal_feasibility_in_rhs_0` in the `global_primal_residual` function when there are box constraints?

https://github.com/Simple-Robotics/proxsuite/blob/c9586f8c5d4bd3368e29138b869521943705d3db/include/proxsuite/proxqp/dense/utils.hpp#L220-L221

Mathematically, my understanding is that this is used for computing the relative errors, and should be

$$
\text{primal\\_feasibility\\_in\\_rhs\\_0} = \max\\{ \lVert C x \rVert\_{\infty}, \lVert x \rVert\_{\infty} \\}
$$

However, the problem right now is that `qpresults.si.tail(qpmodel.dim)` does not contain $x$. It seems like this is not assigned in this function, and either contains zero or some other expression, i.e., from here

https://github.com/Simple-Robotics/proxsuite/blob/c9586f8c5d4bd3368e29138b869521943705d3db/include/proxsuite/proxqp/dense/solver.hpp#L980-L981
This seems like a typo?

Perhaps it should instead be
```c++
primal_feasibility_in_rhs_0 = std::max(
      primal_feasibility_in_rhs_0, infty_norm(qpwork.primal_residual_in_scaled_up.tail(qpmodel.dim)));
```
which does contain $\lVert x \rVert\_{\infty}$, as assigned from earlier in the function?

https://github.com/Simple-Robotics/proxsuite/blob/c9586f8c5d4bd3368e29138b869521943705d3db/include/proxsuite/proxqp/dense/utils.hpp#L196-L200

**Comment by Bambade** [comment]
Dear @oswinso,
Thanks a lot for reporting this typo. ``qpresults.si.tail(qpmodel.dim)`` was indeed not assigned at this stage.  
There are several ways for computing relative errors. One [usual way](https://web.stanford.edu/~boyd/papers/pdf/osqp.pdf) (adapted to the box constraint case) is to indeed use $\||x\||_\infty$ and also the norm of its projection onto the box constraints. I have added these two in PR #258.

**Comment by jcarpent** [comment]
Solved via #258.
